cmd/local.go

```diff
@@ -0,0 +1,367 @@
+
+//Prepare is basically the old EstimateModel function. Responsible for creating directories and preparation.
+func (l LocalModel) Prepare(channels *turnstile.ChannelMap) {
+
+	//Mark the model as started some work
+	channels.Working <- 1
+
+	fs := afero.NewOsFs()
+
+	//Does output directory exist?
+	if ok, _ := afero.Exists(fs, l.Nonmem.OutputDir); ok {
+		//If so are we configured to overwrite?
+		if l.Nonmem.Settings.Overwrite {
+			err := fs.RemoveAll(l.Nonmem.OutputDir)
+			if err != nil {
+				channels.Failed <- 1
+				channels.Errors <- turnstile.ConcurrentError{
+					RunIdentifier: l.Nonmem.Model,
+					Error:         err,
+					Notes:         "An error occured trying to remove the directory as specified in the overwrite flag",
+				}
+				return
+			}
+		} else {
+			channels.Failed <- 1
+			channels.Errors <- turnstile.ConcurrentError{
+				RunIdentifier: l.Nonmem.Model,
+				Error:         errors.New("The output directory already exists"),
+				Notes:         fmt.Sprintf("The target directory, %s already, exists, but we are configured to not overwrite. Invalid configuration / run state", l.Nonmem.OutputDir),
+			}
+			return
+		}
+	}
+
+	//Copy Model into destination and update Data Path
+	err := copyFileToDestination(l.Nonmem, true)
+
+	//Now that the directory is created, let's create the gitignore file if specified
+	if viper.GetBool("git") {
+		WriteGitIgnoreFile(l.Nonmem.OutputDir)
+	}
+
+	if err != nil {
+		channels.Failed <- 1
+		channels.Errors <- turnstile.ConcurrentError{
+			RunIdentifier: l.Nonmem.Model,
+			Error:         err,
+			Notes:         fmt.Sprintf("There appears to have been an issue trying to copy %s to %s", l.Nonmem.Model, l.Nonmem.OutputDir),
+		}
+		return
+	}
+
+	//Create Execution Script
+	scriptContents, err := generateScript(nonMemExecutionTemplate, l.Nonmem)
+
+	if err != nil {
+		channels.Failed <- 1
+		channels.Errors <- turnstile.ConcurrentError{
+			RunIdentifier: l.Nonmem.Model,
+			Error:         err,
+			Notes:         "An error occurred during the creation of the executable script for this model",
+		}
+	}
+
+	//rwxr-x---
+	afero.WriteFile(fs, path.Join(l.Nonmem.OutputDir, l.Nonmem.FileName+".sh"), scriptContents, 0750)
+}
+
+//Work describes the Turnstile execution phase -> IE What heavy lifting should be done
+func (l LocalModel) Work(channels *turnstile.ChannelMap) {
+	cerr := executeNonMemJob(executeLocalJob, l.Nonmem)
+
+	if cerr.Error != nil {
+		recordConcurrentError(l.Nonmem.Model, cerr.Notes, cerr.Error, channels)
+	}
+
+}
+
+//Monitor is unimplemented here. It's the 3rd phase of Turnstile execution
+func (l LocalModel) Monitor(channels *turnstile.ChannelMap) {
+	//Do nothing for this implementation
+}
+
+//Cleanup is the last phase of execution, in which computation / hard work is done and we're cleaning up leftover files, copying results around et all.
+func (l LocalModel) Cleanup(channels *turnstile.ChannelMap) {
+	time.Sleep(10 * time.Millisecond)
```

Comment:
why sleep 10 ms?

Reply:
Was having issues with IO not actually happening consistently. IE when you 30-40 models running, what I would notice is that, without some kind of air gapping, some directories would cleanup and others wouldn't. More work was done afterwards to deal with it, so I can remove this and see if I still have issues.